CMakeLists.txt

```diff
@@ -980,4 +980,11 @@ if(BUILD_CYTHON_MODULES)
 endif()
 
 # https://github.com/apache/incubator-mxnet/issues/20145
-target_compile_definitions(mxnet PRIVATE NVML_NO_UNVERSIONED_FUNC_DEFS)
+if(UNIX)
+    if(USE_NVML)
+      find_package(NVML)
+      if(NVML_FOUND)
+         target_compile_definitions(mxnet PRIVATE NVML_NO_UNVERSIONED_FUNC_DEFS)
```

Comment:
How about moving this to line 595 above and avoid duplication of the nested ifs?

Reply:
I think we need to add this compile def after https://github.com/apache/incubator-mxnet/blob/2cd1b4666d4e6d5df20bc139012061ef0d2e1a37/CMakeLists.txt#L656-L660. In fact I tried what you suggested and got into this error 
```
[2021-04-08T23:00:06.226Z] CMake Error at CMakeLists.txt:600 (target_compile_definitions):

[2021-04-08T23:00:06.226Z]   Cannot specify compile definitions for target "mxnet" which is not built by

[2021-04-08T23:00:06.226Z]   this project.

```

Reply:
You're right. So you can consider add it after line 660, or move 656-660 to an earlier location in the file. It's not necessary to do it now.